src/fragmentarium/ui/info/GenreSelection.tsx

```diff
@@ -0,0 +1,154 @@
+import { Fragment } from 'fragmentarium/domain/fragment'
+import React, { useEffect, useRef, useState } from 'react'
+import { Button, Overlay, Popover } from 'react-bootstrap'
+import Select from 'react-select'
+import classNames from 'classnames'
+import { usePrevious } from 'common/usePrevious'
+import withData from 'http/withData'
+import { Genre, Genres } from 'fragmentarium/domain/Genres'
+
+type Props = {
+  fragment: Fragment
+  updateGenres: (genres: Genres) => void
+  genreOptions: readonly string[][]
+}
+
+function GenreSelection({
+  fragment,
+  updateGenres,
+  genreOptions,
+}: Props): JSX.Element {
+  const [selected, setSelected] = useState<Genre | undefined>(undefined)
+  const [genres, setGenres] = useState(fragment.genres)
+  const prevGenres = usePrevious(genres)
+  const [isDisplayed, setIsDisplayed] = useState(false)
+  const [isUncertain, setIsUncertain] = useState(false)
+  const target = useRef(null)
+
+  function handleChange(event) {
+    const genre = new Genre(event.value, false)
+    setSelected(genre)
+    if (!genres.isPresent(genre)) {
+      setGenres((genres) => genres.insertWithOrder(genre, genreOptions))
+    } else {
+      const retrievedGenre = genres.find(genre) as Genre
+      setSelected(retrievedGenre)
+      setIsUncertain(retrievedGenre.uncertain)
+    }
+  }
+  useEffect(() => {
+    if (!genres.isEqualTo(prevGenres) && prevGenres !== undefined) {
+      updateGenres(genres)
+    }
+  })
```

Comment:
The update should happen in the change event not on render. Is there an easy fix for this?

Reply:
We decided no easy fix currently.